### DISCUSSION TITLE
How to fix error => utilities.css 'spacing[some key][1.5]' does not exist in your theme config

### DISCUSSION THREAD

**Discussion by tuannda** [tailwindlabs/tailwindcss#8650] [Help]
`utilities.css 'spacing[some key][1.5]' does not exist in your theme config. 'spacing' has the following keys: '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '14', '16', '20', '24', '28', '32', '36', '40', '44', '48', '52', '56', '60', '64', '72', '80', '96', 'px', '0.5', '1.5', '2.5', '3.5'
`

I just has error above, Please help me fix it.

```
"tailwindcss": "^3.1.3",
```

**Comment by sharmi** [comment]
I am running into the same error too.

Error Message CssSyntaxError: frontend/static/css/main.css:3:1: 'spacing[some key][1.5]' does not exist in your theme config. 'spacing' has the following keys: '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '14', '16', '20', '24', '28', '32', '36', '40', '44', '48', '52', '56', '60', '64', '72', '80', '96', 'px', '0.5', '1.5', '2.5', '3.5'


my tailwind.config.js file

`module.exports = {
    content: ['../loc1/**/*.{html,js,jinja}', '../../loc2/**/*.{html,js,jinja}'],
    plugins: []
}`

postcss.config.js

`module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {}
  }
}

`
main.css

`    @tailwind base; 
     @tailwind components;
     @tailwind utilities;
`

package.json

   {
  "name": "frontend",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "theme:dev": "postcss static/css/main.css -w -d static/dist/css/ --verbose",
    "theme:build": "NODE_ENV=production postcss static/css/main.css -d static/dist/css/ --verbose"
      },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "amqplib": "^0.10.0",
    "autoprefixer": "^10.4.7",
    "cssnano": "^5.1.11",
    "postcss": "^8.4.14",
    "postcss-cli": "^9.1.0",
    "uuid": "^8.3.2",
    "ws": "^8.8.0"
  },
  "devDependencies": {
    "tailwindcss": "^3.1.3"
  }
    }`

**Reply by Kingscliq** [reply]
Hi just copy the tailwind config file specified in the tailwind documentation, had the same issues and that worked for me 
module.exports = {
  content: [
    "./pages/**/*.{js,ts,jsx,tsx}",
    "./components/**/*.{js,ts,jsx,tsx}",
  ],
  theme: {
    extend: {},
  },
  plugins: [],
}

**Comment by sharmi** [comment]
@Kingscliq Thank you for your response.

I followed your reply. Faced the same error. The only thing different was the content section. 
That led me to believe something in the 'content' paths tripped tailwind. I removed all extraneous paths and restricted it to only html and js folders and that solved the problem. 

For anyone else who face this issue, maybe this will help. pl ensure you do not have non-html or non-js paths (json, python files etc) as part of your content. Postcss/tailwind might read tokens in those files as css lines and try to intepret that leading to errors.

**Reply by M-Jeremy** [reply]
The npx command was checking in other files with this (obv) : 
```
    module.exports = {
      content: ["./**/*.{html,js}"],
      presets: [],
      ....
```
So (its a little projet) i have all my files in one folder and i just changed the "content" line (in tailwind.config.js) to this :
```
    module.exports = {
      content: ["./*.{html,js}"],
      presets: [],
      ....
```
No error and a faster rebuilding (for my little projet)

**Reply by mledwards** [reply]
The only thing to be careful of here is this `content` array will look in `tailwind.config.js` at the top level, which is what was causing my error.

**Reply by davido242** [reply]
Thanks man, this adjustment worked for me too!
I added " /**  ", double asterics as specified from the docs which points to normal directories hitherto it was something like a react app. I am just pointing to a simple html page too for a small webpage! Thanks once more!!

**Reply by Thearbaazshaikh98** [reply]
thank you so much

**Comment by zoltanszogyenyi** [comment]
Same problem here.

**Comment by matthiaskarl91** [comment]
Hey! 

for me extending the tailwind.config.js with 
```
      spacing: {
        'some key': {
          1.5: '<something>', // not sure about the correct value
        },
      },
```

did the trick for me. Seems like it got lost in some update (? 🤔 ). I don't remember changing something in that regard but there were some hints in the error output which relied on that object. Hope it helps someone, too :)

**Reply by crswll** [reply]
You're scanning the `node_modules` directory. You can remove that and adjust the globs in the `content` array of your `tailwind.config.js` file to not include `node_modules`. You should also get a really nice speed up when you build.

**Reply by matthiaskarl91** [reply]
True! Thank you for the hint. But if I remove the node_module path to my component library it somehow doesn't get my styles from that library. I think my setup is kinda off here and I need to find a way how to integrate two separate tailwind apps without collisions 😅

**Reply by migsilva89** [reply]
Worked for me, thanks!

**Reply by danielehrhardt** [reply]
`content: ["./**/*.{html,js,liquid}", "!./**/node_modules/**"],`

**Comment by hugo-abdou** [comment]
same error

**Comment by crswll** [comment]
This happens when `node_modules` is included in the `content` array of your `tailwind.config.js`. Make sure none of the globs in that array capture `node_modules`.

**Comment by mledwards** [comment]
In my case I was being a bit too liberal with what files I was looking in `content: ["./**/*.{html,js}"]` which was scanning the tailwind.config.js too (meta!). I changed this to just the html `content: ["./**/*.html"]` (I'm building out static templates first), and it worked.

**Comment by pwang2** [comment]
If you have a Mono repo, be careful the content glob includes your node_module folder in your leaf packages. :)

**Comment by tybossog** [comment]
I also had same problems, I suggest you go through this specific documentation  as it explains and provides a better understanding to the problem. https://tailwindcss.com/docs/content-configuration. And also a lot of the suggestions here are not the ideal solution, so take the pain and go through it .

**Comment by davidwieler** [comment]
Had this issue as well, for me the fix was updating the globs like @pwang2 said (I'm in a monorepo) to:
```
	content: [
		// app content
		`src/**/*!(node_modules)/*.{ts,svelte,html}`,
		// include packages if not transpiling
		'../../packages/ui/**/*!(node_modules)/*.{ts,svelte}'
	],
```

Fixed the issue and sped up boot time!

**Comment by sriramgroot** [comment]
For me, this issue got resolved by removing the **node_modules** folder and by regenerating the node_modules from **"npm install"**.
